program/src/processor.rs

```diff
@@ -406,11 +418,19 @@ fn process_switch_vote(
             return Err(ProgramError::UninitializedAccount);
         }
 
+        // Jon: I think we might be ok without it technically, but to be safe, can we
+        // have a discriminator check here?
```

Comment:
It's done up in `check_proposal_exists`.

Reply:
Ah ok great! I got confused because there's no discriminator check in `check_governance_exists`

Reply:
No worries, that's bc governance doesn't have a discriminator. The only reason proposals do is because they're not PDAs.

Reply:
Oops, shows how closely I was looking :disappointed:

Reply:
But hold on, that seems even more confusing -- why do some accounts have discriminators and others don't?